projects/graph/src/graph.py

```diff
@@ -2,8 +2,106 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 
+from draw import BokehGraph
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
+
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+
+    def add_edge(self, start, end, bidirectional=True):
+        """Add an edge from start to end."""
+        if start not in self.vertices or end not in self.vertices:
+            raise Exception("Error - vertices not in graph!")
+        self.vertices[start].add(end)
+        if bidirectional:
+            self.vertices[end].add(start)
+    """
+    DFS(graph):
+    for v of graph.verts:
+        v.color = white
+        v.parent = null
+
+    for v of graph.verts:
+        if v.color == white:
+            DFS_visit(v)
+    """
+
+    def dft(self, starting_node, visited=None):
+        """
+        Depth first traversal using recursion
+        """
+        # Mark the node as visited
+        if visited is None:
+            # quese of visited nodes
+            visited = []
+        visited.append(starting_node)
+        # For each child, if that child hasn't been visited, call dft() on that node
+        for node in self.vertices[starting_node].edges:
+            if node not in visited:
+                self.dft(node, visited)
+        return visited   
+
+    """
+    BFS(graph, startVert):
```

Comment:
Multi-line comments are a great way to keep track of what you're doing and document what your code is up to. This is a great habit to have. 👍